web/html/src/manager/virtualization/guests/virtualization-guest-definition-api.js

```diff
@@ -0,0 +1,57 @@
+// @flow
+const React = require('react');
+const Network = require('utils/network');
+const MessagesUtils = require('components/messages').Utils;
+
+declare function t(msg: string, ...args: Array<any>): string;
+
+type Props = {
+  hostid: string,
+  guestUuid: string,
+  children: Function,
+};
+
+type State = {
+  messages: Array<string>,
+  definition: Object,
+};
+
+class VirtualizationGuestDefinitionApi extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      messages: [],
+      definition: {},
+    };
+  }
+
+  componentDidMount() {
+    if (this.state.definition === { }) {
```

Comment:
does this check really work? {} will always be a new object, so this check will fail.

you can use `Object.keys(object).length === 0`

Anyway do you really need this check? componentDidMount will only run once and definition is initialized with {}

Reply:
I changed all the definition checks since that one (and a few others) were causing troubles

Reply:
no longer applicable after numerous refactoring